File: Core/Entities/Post.cs
using Core.Entities.Identity;

namespace Core.Entities
{
    public class Post : BaseEntity
    {
        public string Title { get; set; }

        // Păstrăm Content pentru compatibilitate cu articolele vechi și pentru search/sugestii.
        public string Content { get; set; }

        // Imagine de copertă pentru cardul de blog
        public string ImageUrl { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public string AppUserId { get; set; }
        public AppUser AppUser { get; set; }

        public ICollection<Comment> Comments { get; set; } = new List<Comment>();

        // NOU: secțiuni cu text + imagini multiple
        public ICollection<PostSection> Sections { get; set; } = new List<PostSection>();
    }
}

File: client/src/app/shared/models/post.ts
export interface PostSection {
  id: number;
  heading: string;
  text: string;
  imageUrl?: string;
  caption?: string;
  displayOrder: number;
}

export interface Post {
  id: number;
  title: string;

  // Păstrat pentru articole vechi/search
  summary?: string;
  content: string;

  // Cover image
  imageUrl?: string;

  authorName: string;
  createdAt: string;

  // NOU: secțiuni cu imagini multiple
  sections?: PostSection[];
}